umap/views.py

```diff
@@ -773,13 +781,51 @@ def is_unmodified(self):
                 modified = False
         return modified
 
+    def merge(self):
+        """Try to merge conflicts."""
+
+        # Reference data source of the edition.
+        modified_at = self.request.META.get("HTTP_IF_UNMODIFIED_SINCE")
+
+        for name in self.object.get_versions():
+            path = os.path.join(settings.MEDIA_ROOT, self.object.get_version_path(name))
+            if modified_at == self.compute_last_modified(path):
+                with open(path) as f:
+                    reference = json.loads(f.read())
+                break
+        else:
+            # No reference version found, can't merge.
+            return False
+
+        # New data received in the request.
+        entrant = json.loads(self.request.FILES["geojson"].read())
+
+        # Latest known version of the data.
+        with open(self.path) as f:
+            latest = json.loads(f.read())
+
+        try:
+            merge_conflicts(
```

Comment:
I would better understand if we reuse the value returned by `merge_conflict` likewise:
```suggestion
            merged_features = merge_conflicts(…)
            latest["features"] = merged_features
```

It's a bit more explicit than the in place modification of latest. Maybe `merge_features()` better describe what it does too?

Reply:
> Maybe merge_features() better describe what it does too?

Renamed!

> I would better understand if we reuse the value returned by merge_conflict

If I've correctly understood what you mean, we'll end up with something like:

```diff
diff --git a/umap/views.py b/umap/views.py
index 432e36f..8b0fdd7 100644
--- a/umap/views.py
+++ b/umap/views.py
@@ -794,11 +794,13 @@ class DataLayerUpdate(FormLessEditMixin, GZipMixin, UpdateView):
             latest = json.loads(f.read())
 
         try:
-            merge_features(
+            merged_features = merge_features(
                 reference["features"], latest["features"], entrant["features"]
             )
         except ValueError:
             return False
+        else:
+            latest["features"] = merged_features
 
         # Update request data.
         self.request.FILES["geojson"].file = BytesIO(json.dumps(latest).encode("utf-8"))
```

But then this line is really weird, because those two values are already the exact same:

    +            latest["features"] = merged_features


So I'm not sure what is best :/